### PR TITLE
[docs] Convert docstrings in `_collection.py` to Google style

### PR DESCRIPTION
**Pull request by ebezzi**
**Issue and/or context:**
https://github.com/single-cell-data/TileDB-SOMA/issues/977

**Changes:**
- Convert Python docstrings into [Google-style docstrings](https://google.github.io/styleguide/pyguide.html#38-comments-and-docstrings) 

**Notes for Reviewer:**
1. For now, this only converts _collection.py. As I get feedback I will finish converting the remaining ones.
2. Do we want to regenerate the docs (at least locally) to see if this introduces formatting/design issues?

### CODE REVIEW

File: doc/source/conf.py [hunk 1]
```diff
@@ -40,7 +40,7 @@
 # Add any Sphinx extension module names here, as strings. They can be
 # extensions coming with Sphinx (named 'sphinx.ext.*') or your custom
 # ones.
-extensions = ["sphinx.ext.autodoc", "sphinx.ext.doctest", "sphinx.ext.intersphinx"]
+extensions = ["sphinx.ext.autodoc", "sphinx.ext.doctest", "sphinx.ext.intersphinx", 'sphinx.ext.napoleon']
```
Comment (ebezzi) [inline doc/source/conf.py:43]: `napoleon` is a parser that allows to generate docs from Google's style docstrings.

**Comment by ebezzi** [comment]
Here are a few examples of how the docs will look into Sphinx once converted:

image

image

**Review by johnkerl — APPROVED** [review]
🚢

**Review by bkmartinjr — APPROVED** [review]
LGTM!

**Pull request merged**